enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/Steps.java

```diff
@@ -177,16 +147,31 @@ public NextStepProvider build() {
             }
         }
     }
-
+    
+    /**
+     * Provides values for steps in the multi step dialog.
+     * 
+     */
+    public interface Values {
+        
+        /**
+         * Returns a value for a given {@link Step}.
+         * @param step
+         * @return 
+         */
+        public Object getValueForStep(Class<? extends Step> step);
```

Comment:
If this is declared as 
```
public <X> X getValueForStep(Class<? extends Step<X>> step);
```
then `Steps` clients do not need to do typecasts when extracting the data, if the individual Step classes are properly typed. Isn't it too screwed use of generics @jlahoda  ?